mobile/lib/pages/editing/edit.page.dart

```diff
@@ -85,44 +74,42 @@ class EditImagePage extends ConsumerWidget {
               Navigator.of(context).popUntil((route) => route.isFirst),
         ),
         actions: <Widget>[
-          if (image != null)
-            TextButton(
-              onPressed: () async {
-                try {
-                  final Uint8List imageData = await _imageToUint8List(image!);
-                  ImmichToast.show(
-                    durationInSecond: 3,
-                    context: context,
-                    msg: 'Image Saved!',
-                    gravity: ToastGravity.CENTER,
-                  );
-
-                  await PhotoManager.editor.saveImage(
-                    imageData,
-                    title: '${asset!.fileName}_edited.jpg',
-                  );
-                  await ref.read(albumProvider.notifier).getDeviceAlbums();
-                  Navigator.of(context).popUntil((route) => route.isFirst);
-                } catch (e) {
-                  ImmichToast.show(
-                    durationInSecond: 6,
-                    context: context,
-                    msg: 'Error: ${e.toString()}',
-                    gravity: ToastGravity.BOTTOM,
-                  );
-                }
-              },
-              child: Text(
-                'Save to gallery',
-                style: Theme.of(context).textTheme.displayMedium,
+          TextButton(
+            onPressed: isEdited
```

Comment:
Can you help refactor the logic to its own method?

Reply:
Done should be good now